src/gui/titlebar.cpp

```diff
@@ -0,0 +1,878 @@
+			continue;
+		}
+
+		if (iequals(setting, "program=none")) {
+			maybe_warn_double_program();
+			config.program = TitlebarConfig::ProgramDisplay::None;
+			continue;
+		}
+
+		if (iequals(setting, "program=name")) {
+			maybe_warn_double_program();
+			config.program = TitlebarConfig::ProgramDisplay::Name;
+			continue;
+		}
+
+		if (iequals(setting, "program=path")) {
+			maybe_warn_double_program();
+			config.program = TitlebarConfig::ProgramDisplay::Path;
+			continue;
+		}
+
+		if (iequals(setting, "program=segment")) {
+			maybe_warn_double_program();
+			config.program = TitlebarConfig::ProgramDisplay::Segment;
+			continue;
+		}
+
+		if (iequals(setting, "dosbox")) {
+			maybe_warn_double_dosbox();
+			config.show_dosbox = true;
+			continue;
+		}
+
+		if (iequals(setting, "version")) {
+			maybe_warn_double_version();
+			config.version = TitlebarConfig::VersionDisplay::Default;
+			continue;
+		}
+
+		if (iequals(setting, "version=none")) {
+			maybe_warn_double_version();
+			config.version = TitlebarConfig::VersionDisplay::None;
+			continue;
+		}
+
+		if (iequals(setting, "version=simple")) {
+			maybe_warn_double_version();
+			config.version = TitlebarConfig::VersionDisplay::Simple;
+			continue;
+		}
+
+		if (iequals(setting, "version=detailed")) {
+			maybe_warn_double_version();
+			config.version = TitlebarConfig::VersionDisplay::Detailed;
+			continue;
+		}
+
+		if (iequals(setting, "cycles")) {
+			maybe_warn_double_cycles();
+			config.show_cycles = true;
+			continue;
+		}
+
+		if (iequals(setting, "mouse")) {
+			maybe_warn_double_mouse();
+			config.mouse = TitlebarConfig::MouseHintDisplay::Default;
+			continue;
+		}
+
+		if (iequals(setting, "mouse=none")) {
+			maybe_warn_double_mouse();
+			config.mouse = TitlebarConfig::MouseHintDisplay::None;
+			continue;
+		}
+
+		if (iequals(setting, "mouse=short")) {
+			maybe_warn_double_mouse();
+			config.mouse = TitlebarConfig::MouseHintDisplay::Short;
+			continue;
+		}
+
+		if (iequals(setting, "mouse=full")) {
+			maybe_warn_double_mouse();
+			config.mouse = TitlebarConfig::MouseHintDisplay::Full;
+			continue;
+		}
+
+		LOG_WARNING("SDL: Invalid 'window_titlebar' setting '%s', ignoring",
+		            setting.c_str());
+		config_needs_sync = true;
+	}
+
+	if (config_needs_sync) {
+		sync_config();
+	}
+}
+
+void TITLEBAR_ReadConfig(const Section_prop& secprop)
+{
+	parse_config(secprop.Get_string("window_titlebar"));
+
+	GFX_RefreshTitle();
+}
+
+void TITLEBAR_AddConfig(Section_prop& secprop)
+{
+	constexpr auto always = Property::Changeable::Always;
+
+	Prop_string* prop_str = nullptr;
+
+	prop_str = secprop.Add_string("window_titlebar",
+	                              always,
+	                              "program=name dosbox cycles mouse=full");
+	prop_str->Set_help(
+	        "Space separated list of information to be displayed on the window title bar\n"
+	        "('program=name dosbox cycles mouse=full' by default).\n"
```

Comment:
This is implied, but I think we should state it explicitly: 

`If a parameter is not specified, its default value is used.`

Reply:
OK, added.